source/appModules/devenv.py

```diff
@@ -473,3 +486,36 @@ class IVsTextManager(comtypes.IUnknown):
     COMMETHOD([], HRESULT, 'SetTopLine',
               ( ['in'], c_int, 'iBaseLine' )),
 ]
+
+class UIATextInfoForVisualStudio( UIATextInfo):
+
+	def __init__(self,obj,position,_rangeObj=None):
+		super(UIATextInfoForVisualStudio,self).__init__(obj,position,_rangeObj)
+		self.digitsAtStartOfLineMatch = re.compile(r"^[\d]+\s")
+
+	def _getFormatFieldAtRange(self,range,formatConfig):
+		formatField = super( UIATextInfoForVisualStudio , self)._getFormatFieldAtRange(range,formatConfig)
+		currentElement = self.NVDAObjectAtStart
+		isLineNumberAtStart = self.digitsAtStartOfLineMatch.match(range.GetText(-1))
+		if isLineNumberAtStart: 
+			currentLineNumber = int(isLineNumberAtStart.group() )
+			breakPoint = self._getBreakPointAtLine( currentLineNumber , currentElement )
+			if breakPoint: 
+				formatField.field["breakpoint"] = breakPoint
+		return formatField 
+
+	def _getBreakPointAtLine(self,line,textObject):
+		if textObject and textObject.next and textObject.next.name == "Glyph Margin Grid":
+			currentBreakPoint = textObject.next.firstChild 
+			while currentBreakPoint: 
+				currentUIAutomationId = currentBreakPoint.UIAElement.currentAutomationID
+				currentLineNumber = int(currentUIAutomationId.split(';')[1])
+				if currentLineNumber == line: return currentBreakPoint.name.split(';')[0]
+				currentBreakPoint = currentBreakPoint.next 

```

Comment:
I think it might be cheaper to call .children rather than .firstChild and repeated .next, as .children will fetch most of what you need in one cross-process call, rather than several. Also, you can use CachedAutomationId instead of CurrentAutomationId. @MichaelDCurran, would you agree here?